larray/inout/xw_reporting.py

```diff
@@ -0,0 +1,709 @@
+
+        >>> # add graph to the output sheet 'Belgium'
+        >>> sheet_be.add_graph('Population', demo.pop['Belgium'], template='Line')
+        """
+        pass
+
+    def sheet_names(self):
+        r"""
+        Returns the names of the output sheets.
+
+        Examples
+        --------
+        >>> report = ExcelReport()
+        >>> sheet_pop = report.new_sheet('Pop')
+        >>> sheet_births = report.new_sheet('Births')
+        >>> sheet_deaths = report.new_sheet('Deaths')
+        >>> report.sheet_names()
+        ['Pop', 'Births', 'Deaths']
+        """
+        pass
+
+    def to_excel(self, filepath, data_sheet_name='__data__', overwrite=True):
+        r"""
+        Generate the report Excel file.
+
+        Parameters
+        ----------
+        filepath : string
+            Path of the report file for the dump.
+        data_sheet_name : string, optional
+            name of the Excel sheet where all data associated with items is dumped.
+            Defaults to '__data__'.
+        overwrite : bool, optional
+            whether or not to overwrite an existing report file.
+            Defaults to True.
+
+        Examples
+        --------
+        >>> demo = load_example_data('demo')
+        >>> templates_dir = get_example_templates_dir()
+        >>> report = ExcelReport(templates_dir)
+        >>> report.template = 'Line_Marker'
+
+        >>> for c in demo.country:
+        ...     sheet_country = report.new_sheet(c)
+        ...     sheet_country.add_graph('Population', demo.pop[c])
+        ...     sheet_country.add_graph('Births', demo.births[c])
+        ...     sheet_country.add_graph('Deaths', demo.deaths[c])
+
+        Basic usage
+
+        >>> report.to_excel('Demography_Report.xlsx')
+
+        Alternative data sheet name
+
+        >>> report.to_excel('Demography_Report.xlsx', data_sheet_name='Data Tables') # doctest: +SKIP
+
+        Check if ouput file already exists
+
+        >>> report.to_excel('Demography_Report.xlsx', overwrite=False) # doctest: +SKIP
+        Traceback (most recent call last):
+        ...
+        ValueError: Sheet named 'Belgium' already present in workbook
+        """
+        pass
+
+
+if xw is not None:
+    from xlwings.constants import LegendPosition, HAlign, VAlign, ChartType, RowCol
+    from larray.inout.xw_excel import open_excel
+
+    class ItemSize(object):
+        def __init__(self, width, height):
+            self.width = width
+            self.height = height
+
+        @property
+        def width(self):
+            return self._width
+
+        @width.setter
+        def width(self, width):
+            _positive_integer(width)
+            self._width = width
+
+        @property
+        def height(self):
+            return self._height
+
+        @height.setter
+        def height(self, height):
+            _positive_integer(height)
+            self._height = height
+
+    class AbstractExcelItem(ItemSize):
```

Comment:
In my **very subjective** opinion (so this is your call), this seems like one class too much in the inheritance hierarchy. High depth inheritance has a cost (both in terms of speed -- which I don't care about here -- and in the time it takes to understand the code). Yes, this is a trade-off but well... I would rather have this class inlined into ExcelTitleItem and ExcelGraphItem.